### ISSUE TITLE
Feature Request: Windows Hello support

### ISSUE THREAD

**Issue by Gibletron** [keepassxreboot/keepassxc#2462]
On my Android I use Keepass2Android with support for fingerprint database unlocking.
I would like to request this same functionality on Windows PC's with Hello support.
Windows Hello is a bio metric safety feature used to unlock Windows using for instance a fingerprint or facial recognition.

It would be great to be able to unlock my Keepass with a swipe of my fingerprint sensor, without having to type in a password or anything of the sorts.
Could also be used as an MFA feature, where Windows Hello authentication is required in addition to other authentication methods

**Comment by droidmonkey** [comment]
I like your proposal, but only for supporting #488.

Here is the ugly truth about fingerprint and other biometrics. They are NOT a password! They are your USERNAME. Biometrics "prove" that you are who you say you are. They DO NOT prove what you know (ie, your password). KeePass2Android seemingly uses your biometrics to encrypt your password and store it in the app data. When you present your fingerprint, it decrypts the password using the fingerprint hash (most likely) and types it into the password field.

**Comment by droidmonkey** [comment]
The equivalent on PC is a TPM chip, which is a hardware based encryption and key storage device soldered to your motherboard. It could be possible to use Windows Hello to authenticate a request to the TPM chip which would give use back the credentials for your database, similar to how KeePass2Android works.

**Comment by tinokorth** [comment]
I'd also love the integration of Windows Hello.

**Comment by mgabi96** [comment]
There is a plugin for windows hello with keepass 2. It would be great to have this feature in keepassxc

**Comment by karlhorky** [comment]
I suppose this feature is similar to the one that was added (but not yet released) to add TouchID support on macOS:

[Add support for quick unlock with TouchID on Macbook Pro (#1851)](https://github.com/keepassxreboot/keepassxc/pull/1851)

**Comment by optomux** [comment]
What's the status on this? Fingerprint support would be great.

**Comment by MisterY** [comment]
As I'm currently interested in Windows implementation (where the database locks with the workstation and I have to unlock it many times a day), here are some references from KeePass plugins:

https://github.com/sirAndros/KeePassWinHello
https://github.com/Angelelz/WinHelloUnlock

Even though these are written in dotNet, hope this can be useful as an API reference.

**Comment by userosos** [comment]
@MisterY how it works for keepassxc. It be will work in keepassxc now? I want use my fingerprint device on my laptop now! I think it not work now.

**Comment by droidmonkey** [comment]
This is not currently an available feature in KeePassXC. I am going to make an attempt to integrate this along with Quick Unlock for 2.6.0.

**Comment by janjur** [comment]
@droidmonkey, this will be much appreciated :)

**Comment by Mardiie** [comment]
Just uninstalled KeePass to read this.
Must have feature.

**Comment by mgabi96** [comment]
This is the only missing feature for me. For a long time i want to switch from keepass to Keepassxc

**Comment by droidmonkey** [comment]
Ya'll gonna be waiting forever because it's not looking possible. Windows Hello is for UWP apps only.

**Comment by Gibletron** [comment]
> Ya'll gonna be waiting forever because it's not looking possible. Windows Hello is for UWP apps only.

I am not a programmer in any way, but there is a keepass2 plugin that as windows Hello support to keepass2..
As far as I know keepass2 is not a UWP
Do not sure how they do it, bit it might just be possible.

Then again, it doesn't have any priority for me, since I have moved to bitwarden

**Comment by phoerious** [comment]
Keepass2 is C#.

**Comment by Mardiie** [comment]
> 
> 
> Ya'll gonna be waiting forever because it's not looking possible. Windows Hello is for UWP apps only.

100% not true.
KeePass, what this software is based on, has 2 plugins that support it, and that it is not even near an UWP app.

**Comment by phoerious** [comment]
> Written in, then compiled, just like any other (Windows) program.

Wrong. Not like ANY other. C# is still .NET and the UWP/WINRT APIs are .NET APIs and not available in C++ except for maybe with cppwinrt.

**Comment by Mardiie** [comment]
> 
> 
> > Written in, then compiled, just like any other (Windows) program.
> 
> Wrong. Not like ANY other. C# is still .NET and the UWP/WINRT APIs are .NET APIs and not available in C++ except for maybe with cppwinrt, which is still problematic without the VCpp toolchain. We can of course load DLLs, but we certainly don't want to GetProcAddress() the whole WINRT API.

So whats your saying is it can't be coded?
I find that... unique.
https://github.com/sirAndros/KeePassWinHello/issues/7#issuecomment-525311371
That would be highly problematic for KeepassXC since Windows is going UWP all the way.

It's fine by me either way.

**Comment by phoerious** [comment]
The library you linked is a C# library. 🙄

**Comment by mgabi96** [comment]
What about the microsoft biometrics framework wich is still used in LastPass (even if we cant see how) ?

**Comment by Mardiie** [comment]
> 
> 
> The library you linked is a C# library. 🙄

Wow, reading a post is hard.
He is stating he will write a library for C++/C#.
Nice try to make it seem you aren't wrong though.

**Comment by phoerious** [comment]
Thank you for your insightful contribution. Come back when you have a C++ interface for us.
